### PR TITLE
Track some changes in 2020/02/02 release

### PR DESCRIPTION
**Pull request by dr-scsi**
Dear all,

this is a first attempt to update fntguide.tex to catch up with LaTeX 2020/02/02 release.  Some points are still missing, but I want to share this part with the team in order to receive some feedback before proceeding.  Hence, any comments welcome.

### CODE REVIEW

File: base/doc/fntguide.tex [hunk 1]
```diff
@@ -489,14 +504,26 @@ \subsection{Parameters for author commands}
...
+\NEWfeature{2020/02/02}
 The shapes selected by |\textit|, |\itshape|, |\textsl|, |\slshape|,
-|\textsc|, |\scshape|, |\textup| and |\upshape|.  By default these are
-|it|, |sl|, |sc| and |n|.
+|\textsc|, |\scshape|, |\textssc|, |\sscshape|, |\textsw|, |\swshape|,
+|\textulc|, |\ulcshape|, |\textup| and |\upshape|.  By default these are
+|it|, |sl|, |sc|, |ssc|, |sw|, |ulc| and |n|.  |ulc| is special here
+since it is a virtual one; it doesn't exist as a real shape.  The
+respective macros |\textulc| or |\ulcshape| change small capitals back
+to upper/lower case which in turn will not change the font with respect
```
Comment (FrankMittelbach) [inline base/doc/fntguide.tex:522]: but will not change ...

File: base/doc/fntguide.tex [hunk 2]
```diff
@@ -489,14 +504,26 @@ \subsection{Parameters for author commands}
...
+to italics, slanted or swash.  |\upshape| or |\textup| behave
+differently here and switch italics or slanted back to upright.
```
Comment (FrankMittelbach) [inline base/doc/fntguide.tex:524]: \upshape or textup in contrast will switch italics or slanted back to upright but to not alter the state of upper/lower case, e.g., keep small capitals if present.

File: base/doc/fntguide.tex [hunk 3]
```diff
@@ -1874,6 +1901,66 @@ \subsection{The order of declaration}
...
 
+\subsection{Font series defaults per document family}
+
+\NEWfeature{2020/02/02}
+With additional weights and widths being available in many font families
+nowadays, it is more likely that somebody will want to match, say, a
+medium weight serif family with a semi-light sans serif family, or that
+with one family one wants to use the bold-extended face when |\textbf|
+is used, while with another it should be bold (not extended) or
+semi-bold, etc.  \LaTeX{} stores family specific defaults for series in
```
Comment (FrankMittelbach) [inline base/doc/fntguide.tex:1912]: don't think that the internal macros should be given in the guide, better just describe the high-level interface that should be used.

File: base/doc/fntguide.tex [hunk 4]
```diff
@@ -1874,6 +1901,66 @@ \subsection{The order of declaration}
...
+\begin{decl}
+  |\mdseries@rm| \\
+  |\mdseries@sf| \\
+  |\mdseries@tt| \\
+  |\bfseries@rm| \\
+  |\bfseries@sf| \\
+  |\bfseries@tt|
+\end{decl}
+
+The default values can be overwritten by the author level macro in
```
Comment (FrankMittelbach) [inline base/doc/fntguide.tex:1923]: replace auther level macro with declation

maybe

the default values can be altered using a \cs{DeclareFontSeriesDefault} declaration

File: base/doc/fntguide.tex [hunk 5]
```diff
@@ -1874,6 +1901,66 @@ \subsection{The order of declaration}
...
+packages or document preambles:
+\begin{decl}
+  |\DeclareFontSeriesDefault| \oarg{doc family}
+  \arg{doc series} \arg{series value}
```
Comment (FrankMittelbach) [inline base/doc/fntguide.tex:1927]: my name for this is not "doc family" but  "meta family"

File: base/doc/fntguide.tex [hunk 6]
```diff
@@ -1874,6 +1901,66 @@ \subsection{The order of declaration}
...
+\end{decl}
+This declaration takes three arguments:
+\begin{description}
+\item[Document family interface:] Can be either |rm|, |sf| or |tt|.
+  This is optional and if not given the overall default.
```
Comment (FrankMittelbach) [inline base/doc/fntguide.tex:1932]: 1932: is not a real sentence

**Comment by FrankMittelbach** [comment]
thanks. Actually a few minutes go I  noticed that I also still had this on my todo list and that it is high time to do something about it. Overall looks good, couple of comments added.

**Comment by dr-scsi** [comment]
Many thanks for your comments.  I have incorporated them, I hope I've caught them all.  Again, any comments welcome.

**Comment by FrankMittelbach** [comment]
thanks

**Comment by dr-scsi** [comment]
Thanks for accepting and merging the PR.  I think there are still two topics missing in `fntguide.tex`:

- Providing font family substitutions with `\DeclareFontFamilySubstitution`, and
- `textcomp` macros now being part of the kernel.

Should these topics also be mentioned?  Anything else missing?

**Comment by FrankMittelbach** [comment]
Am 28.06.20 um 21:01 schrieb dr-scsi:
> Thanks for accepting and merging the PR. I think there are still two 
> topics missing in |fntguide.tex|:
> 
>   * Providing font family substitutions with
>     |\DeclareFontFamilySubstitution|, and
>   * |textcomp| macros now being part of the kernel.
> 
> Should these topics also be mentioned? Anything else missing?

I guess both should be documented (perhaps something can be taken from 
ltnews as a basis. Not sure if anything else is missing. Nothing that 
comes to my mind directly, but looking over the recent ltnews should 
verify that (or changes.txt). It should all be covered there.

**Pull request merged**